### PR TITLE
Mesh: implement function Mesh.selectAntenna().

### PR DESCRIPTION
**Pull request by XuGuohui**

### CODE REVIEW

File: wiring/inc/spark_wiring_mesh.h [hunk 1]
```diff
@@ -38,6 +38,11 @@
 
 namespace spark {
 
+typedef enum {
+    MESH_ANT_INTERNAL = 0,
```
Comment (avtolstoy) [inline wiring/inc/spark_wiring_mesh.h]: Let's reserve `0` for default.

File: wiring/inc/spark_wiring_mesh.h [hunk 2]
```diff
@@ -182,6 +187,8 @@ class MeshClass : public NetworkClass, public MeshPublish {
         return network_ready(*this, 0,  NULL);
     }
 
+    void selectAntenna(MeshAntennaType antenna);
```
Comment (avtolstoy) [inline wiring/inc/spark_wiring_mesh.h]: Should return `int` (and the error codes from `system_error_t`).

**Comment by XuGuohui** [comment]
So I'll close this PR and export this function in https://github.com/particle-iot/device-os/pull/1583.

**Comment by XuGuohui** [comment]
It needs to change the Boron system part1 linker script as well.

**Pull request closed without merging**